cg/meta/archive/ddn.py

```diff
@@ -0,0 +1,140 @@
+"""Module for archiving and retrieving folders via DDN."""
+from pathlib import Path
+from typing import Dict, List
+from urllib.parse import urljoin
+from requests.models import Response
+
+from datetime import datetime
+from cg.constants.constants import APIMethods, FileFormat
+from cg.io.controller import APIRequest, ReadStream
+from cg.models.cg_config import DDNConfig
+
+
+class DDNApi:
+    """Class for archiving and retrieving folders via DDN."""
+
+    def __init__(self, config: DDNConfig):
+        self.database_name: str = config.database_name
+        self.user: str = config.user
+        self.password: str = config.password
+        self.url: str = config.url
+        self.archive_repository = config.archive_repository
+        self.local_storage = config.local_storage
+        self.auth_token: str
+        self.refresh_token: str
+        self.token_expiration: datetime
+        self._set_auth_tokens()
+        self.headers: Dict[str, str] = {
+            "Content-Type": "application/json",
+            "accept": "application/json",
+        }
```

Comment:
Should this be a general get function for reuse by other APIs?

Reply:
I could not find anywhere else in cg where this is duplicated, so I say we keep it here for now